File: Subchapter.js
class SubChapter extends TableOfContents {
  constructor(name) {
    super();
    this.name = name;
    this.elements = [];
  }

  addParagraph(text) {
    const Paragraph = require('./Paragraph');
    const p = new Paragraph(text);
    this.elements.push(p);
    return p;
  }

  addImage(imageName) {
    const Image = require('./Image');
    const i = new Image(imageName);
    this.elements.push(i);
    return i;
  }

  addTable(title) {
    const Table = require('./Table');
    const t = new Table(title);
    this.elements.push(t);
    return t;
  }

}

File: src/SubChapter.java
import java.util.ArrayList;
import java.util.List;

public class SubChapter extends TableOfContents {
    String name;
    private final List<TableOfContents> elements = new ArrayList<>();

    public SubChapter(String name) {
        this.name = name;
    }

    public Paragraph addParagraph(String text) {
        Paragraph p = new Paragraph(text);
        elements.add(p);
        return p;
    }

    public Image addImage(String imageName) {
        Image img = new Image(imageName);
        elements.add(img);
        return img;
    }

    public Table addTable(String title) {
        Table t = new Table(title);
        elements.add(t);
        return t;
    }

    public List<TableOfContents> getElements() {
        return elements;
    }
}

